lib/dbservice/groups.ex

```diff
@@ -57,6 +57,19 @@ defmodule Dbservice.Groups do
     Repo.get_by(Group, child_id: child_id, type: "batch")
   end
 
+  @doc """
+  Gets group by child_id and type.
+  Raises `Ecto.NoResultsError` if the Group does not exist.
+  ## Examples
+      iex> get_group_by_child_id_and_type(123,"batch")
+      %Group{}
+      iex> get_group_by_child_id_and_type(456,"batch")
+      ** (Ecto.NoResultsError)
+  """
+  def get_group_by_child_id_and_type(child_id, type) do
+    Repo.get_by(Group, child_id: child_id, type: type)
+  end
+
```

Comment:
This function is already defined above where type is hardcoded. Maybe we can reuse that and also pass "batch" as argument where it was used.

Reply:
Ok, I have done the same in the latest commit.